### DISCUSSION TITLE
RIICL: Rewrite terminal.c in Common Lisp

### DISCUSSION THREAD

**Discussion by jgarte** [lem-project/lem#1558] [Ideas]
# An Idea 💡 

What do you think if instead of using `libvterm` we were to just write that terminal emulation from scratch in Common Lisp?

I think that #1557 should be completed first.

My inspiration for this idea comes from the eat project:

## Related Links

- https://codeberg.org/akib/emacs-eat
- https://akib.codeberg.page/blog/emacs-lisp-is-fast.html
- https://akib.codeberg.page/blog/introducing-eat.html

> Anything Emacs Lisp can do Common Lisp can do better 😆

**Comment by vindarel** [comment]
A work someone needs to do!

**Comment by cxxxr** [comment]
When I was making terminal-mode, it flashed across my mind, but I looked away because it was a lot of work.
But I want it.
